qbraid/runtime/origin/provider.py

```
@@ -0,0 +1,153 @@
+        """Get a list of available OriginQ devices."""
+        catalog = self.service.backends()
+        device_ids: set[str] = set()
+        for backend_id, available in catalog.items():
+            if available is False:
+                continue
+            if hardware_only and backend_id in SIMULATOR_BACKENDS:
+                continue
+            device_ids.add(backend_id)
+        return [self.get_device(device_id) for device_id in sorted(device_ids)]
+
+    def __hash__(self):
```

Ruff (0.15.9)

[warning] Missing return type annotation for special method `__hash__`

(ANN204)